src/main/java/ui/components/pickers/PickerMilestone.java

```diff
@@ -93,6 +96,47 @@ private void setRemovedInUI(Label milestone) {
         milestone.getStyleClass().add("labels-removed"); // add strikethrough
     }
 
+    public static Optional<PickerMilestone> getExistingMilestone(List<PickerMilestone> milestoneList) {
+        return milestoneList.stream()
+                .filter(PickerMilestone::isExisting)
+                .findAny();
+    }
+
+    public static Optional<PickerMilestone> getSelectedMilestone(List<PickerMilestone> milestoneList) {
+        return milestoneList.stream()
+                .filter(PickerMilestone::isSelected)
+                .findAny();
+    }
+
+    /**
+     * Gets the default milestone
+     * If there is an existing milestone, default milestone is the existing milestone
+     * Else it is the first open milestone that is not overdue
+     * Precondition: sortedMilestoneList needs to be sorted in its natural order
+     *
+     * @param sortedMilestoneList
+     * @return
```

Comment:
@HansNewbie It is better if I fill it up? 
`@return Optional of default milestone`